File: core/src/main/java/io/nirahtech/erp/core/Siren.java
package io.nirahtech.erp.core;

public final class Siren {
    private final int code;
    private static final int CODE_LENGTH = 9;

    private Siren(int code) {
        this.code = code;
    }

    public static Siren create(int code) throws IllegalArgumentException {
        if (String.valueOf(code).length() != CODE_LENGTH) {
            throw new IllegalArgumentException(String.format("SIREN must be a number with %s characters.", CODE_LENGTH));
        }
        return new Siren(code);
    }

    public int getCode() {
        return code;
    }

}

File: webapp/src/main/react/src/data/Siren.ts
class Siren {
    private readonly code: number;
    private static readonly CODE_LENGTH: number = 9;
  
    private constructor(code: number) {
      this.code = code;
    }
  
    static create(code: number): Siren {
      if (String(code).length !== Siren.CODE_LENGTH) {
        throw new Error(`SIREN must be a number with ${14} characters.`);
      }
      return new Siren(code);
    }
  
    getCode(): number {
      return this.code;
    }
  }
export default Siren;  